### DISCUSSION TITLE
After creating a clog object (div, a href, etc), how to list / access all objects?

### DISCUSSION THREAD

**Discussion by aykaramba** [rabbibotton/clog#33] [General]
Hi.

I am getting the hang of the basics of creating and rendering objects.  Can someone clue me in to the basic principles on:

1) After I have created a bunch of objects on a page, how do I list them and their properties in the slime repl?

2) After I list created objects how do I reference them to update their properties?  Do they have unique ids?

I am working through the examples slowly and I will need to re-do a bunch of tutorials in Practial Common Lisp obviously.  Being able to create dom objects, list them and then edit them would be a helpful step forward.

Clog is totally fun.  It made me re-start learning Lisp.  

Thanks!

**Comment by rabbibotton** [comment]
1. You would need to bind the objects globally (defvar/defparameter) 

CLOG-USER> (clog-repl)
Use clog-user:*body* to access the clog-repl window.
NIL
New connection id - 3885 - #<a WEBSOCKET-DRIVER.WS.SERVER:SERVER 0x1112cc640>
CLOG-USER> (defvar *my-div*)
*MY-DIV*
CLOG-USER> (setf *my-div* (create-div *body* :content "Hello World"))

2. They have unique ids under the cover but you don't deal with them, they are lisp objects like any other bound to symbols.

**Comment by aykaramba** [comment]
Aha!  Simple.  Thank you very kindly.

**Comment by aykaramba** [comment]
I wanted to drop by and say thank you for cluing me in to operating on lisp objects. I am now going to go back and re-do a bunch of books and tutorials with CLOG.

This is a lot of fun.
